test/e2e/chaos/basic.go

```diff
@@ -1266,6 +1268,156 @@ var _ = ginkgo.Describe("[Basic]", func() {
 		})
 	})
 
+	ginkgo.Context("[Sidecar Config]", func() {
+		var (
+			cmName      string
+			cmNamespace string
+		)
+
+		// delete the created config map in each test case
+		ginkgo.JustAfterEach(func() {
+			kubeCli.CoreV1().ConfigMaps(cmNamespace).Delete(cmName, &metav1.DeleteOptions{})
+		})
+
+		ginkgo.Context("[Template Config]", func() {
+
+			ginkgo.It("[InValid ConfigMap key]", func() {
+				ctx, cancel := context.WithCancel(context.Background())
+				cmName = "incorrect-key-name"
+				cmNamespace = chaosMeshNamespace
+				err := createTemplateConfig(ctx, cli, cmName,
+					map[string]string{
+						"chaos-pd.yaml": `name: chaosfs-pd
+selector:
+  labelSelectors:
+    "app.kubernetes.io/component": "pd"`})
+				framework.ExpectNoError(err, "failed to create template config")
+
+				err = wait.Poll(time.Second, 10*time.Second, func() (done bool, err error) {
+					dpi := kubeCli.AppsV1().Deployments(chaosMeshNamespace)
+					dep, err := dpi.Get(chaosControllerManager, metav1.GetOptions{})
```

Comment:
If you want to ensured that the chaos-manager won't crash because of invalid configmap. I think we should `Get` the original Chaos-mesh controller Pod first and record its UID, and after we apply the invalid configmap, We shall check the controller pod uid again.

Reply:
I think we should both verify the uid and [restartCount](https://github.com/kubernetes/api/blob/54aa84730ddd60fec0ab4325f2b66f8a9790cd98/core/v1/types.go#L2403) for the controller-manager pod

Reply:
Good point. I will change my pr as you said